File: frontend/lib/src/features/ranking/domain/entities/ranked_company.dart
class RankedCompany {
  final String symbol;
  final String name;
  final double intrinsicValue;
  final double currentPrice;
  final double marginOfSafety;

  const RankedCompany({
    required this.symbol,
    required this.name,
    required this.intrinsicValue,
    required this.currentPrice,
    required this.marginOfSafety,
  });
}

File: backend/common/src/main/java/afsdigital/grahamselect/valuation/domain/entities/RankedCompany.java
package afsdigital.grahamselect.valuation.domain.entities;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;

@Builder
@Data
public class RankedCompany {

    private String symbol;
    private String name;
    private BigDecimal intrinsicValue;
    private BigDecimal currentPrice;
    private BigDecimal marginOfSafety;
    private LocalDate intrinsicValueUpdatedAt;

}
